drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,476 @@
+/*
+ * Copyright (C) 2016,2017,2018 Kees Bakker, SODAQ
+ * Copyright (C) 2017 George Psimenos
+ * Copyright (C) 2018 Steffen Robertz
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sht2x
+ * @{
+ *
+ * @file
+ * @brief       Device driver implementation for the SHT2x temperature and
+ *              humidity sensor.
+ *
+ * @author      Kees Bakker <kees@sodaq.com>
+ * @author      George Psimenos <gp7g14@soton.ac.uk>
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ *
+ * @}
+ */
+
+#include <math.h>
+
+#include "log.h"
+#include "sht2x.h"
+#include "sht2x_params.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+/**
+ * @brief       The number of retries when doing a polled measurement
+ */
+#define MAX_RETRIES         20
+
+/**
+ * @brief       A few helper macros
+ */
+#define BUS                 (dev->params.i2c_dev)
+#define ADDR                (dev->params.i2c_addr)
+#define MODE                (dev->params.measure_mode)
+#define CRC                 (dev->params.is_crc_enabled)
+
+typedef enum {
+    eTempHoldCmd        = 0xE3,     /**< trigger temp measurement, hold master */
```

Comment:
From our [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#naming):"Do NOT use CamelCase. Function, variable and file names as well as enums, structs or typedefs are written in lowercase with underscores."

Reply:
These names were copied from the supplier's example code. But OK, I'll change it.